backend/src/import/async/measures/degreeCentrallity.ts

```
@@ -0,0 +1,44 @@
+import ImportAPISpecs from "api-specifications/import";
+import { RowsProcessedStats } from "import/rowsProcessedStats.types";
+
+import { Readable } from "node:stream";
+import { BatchProcessor } from "import/batch/BatchProcessor";
+import { getRepositoryRegistry } from "server/repositoryRegistry/repositoryRegistry";
+import { ISkillConnection } from "esco/occupationToSkillRelation/occupationToSkillRelation.types";
+
+/**
+ * Updates the degree centrality for the given skills. The degree centrality is the number of edges connected to a skill.
+ * @param {ISkillConnection[]} skills - The skills to update.
+ * @returns {Promise<RowsProcessedStats>} - A Promise that resolves to the number of rows processed, the number of rows successfully updated and the number of rows that failed.
+ */
+export async function updateSkillDegreeCentrality(skills: ISkillConnection[]): Promise<RowsProcessedStats> {
+  const response = await getRepositoryRegistry().skill.updateSkillDegreeCentrality(skills);
+
+  return {
+    rowsFailed: response.modifiedCount === skills.length ? 0 : skills.length - response.modifiedCount,
+    rowsSuccess: response.modifiedCount,
+    rowsProcessed: skills.length,
+  };
+}
+
+/**
+ * Calculates the degree centrality for the given modelId.
+ * @param {ImportAPISpecs.Types.POST.Request.Payload} event - The event object containing details for the import process.
+ */
+async function calculateSkillDegreeCentrality(event: ImportAPISpecs.Types.POST.Request.Payload) {
+  const BATCH_SIZE = 1000;
+
```

Shouldn't this be in a try/catch?

---

cant any of these throw? Including tthe batchProcessor?

---

No wait... the batch processor never throws... but the repository methods?

---

or is the idea that we let it bubble up to the measures.ts and catch it there?

---

even so... perhaps we should log and rethrow... since we should test that it WILL throw if the repositories throw and not, maybe hang? (probably wont... but we should have a test, right?)